lib/alerts/de_userfying_edit_alert_monitor.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# This class identifies articles that have been moved from
+# their use space by users
+# Tag: de-userfying
+# Cf. https://en.wikipedia.org/wiki/MediaWiki:Tag-de-userfying
+class DeUserfyingEditAlertMonitor
+  def self.create_alerts_for_deuserfying_edits
+    new.create_alerts
+  end
+
+  def initialize
+    @wiki = Wiki.find_by(language: 'en', project: 'wikipedia')
+  end
+
+  def create_alerts
+    student_edits = edits_made_by_students(edits, current_students)
+    student_edits.each do |edit|
+      article = Article.find_by(mw_page_id: edit['pageid'])
+      user = User.find_by(username: edit['user'])
+      course_ids = courses_for_a_student(user.id)
+      course_ids.each do |course_id|
+        create_alert(user.id, course_id, article.id, edit['revid'])
+      end
+    end
+  end
+
+  def edits
+    api = WikiApi.new @wiki
+    query_params = {
```

Comment:
It would be helpful to include comments to characterize this query, including
* How many edits over roughly how much time result from this query (ie, it looks like there are something like ~20 de-userfying edits on English Wikipedia per day)
* The equivalent URL for doing the same query manually in a browser

Reply:
Added these comments. Hope they are ok.

Reply:
Great. I suggest adding `limit: 100` to the parameters, as there could definitely be spikes in de-userfying edits where more than 10 happen in a short timeframe, so we that might miss some between ConstantUpdate runs.

Reply:
Added the 100 limit